roofit/roofitcore/inc/RooAbsDataStore.h

```diff
@@ -93,14 +98,18 @@ class RooAbsDataStore : public TNamed, public RooPrintable {
     printStream(defaultPrintStream(),defaultPrintContents(options),defaultPrintStyle(options));
   }
 
-  virtual void printName(std::ostream& os) const ;
-  virtual void printTitle(std::ostream& os) const ;
-  virtual void printClassName(std::ostream& os) const ;
-  virtual void printArgs(std::ostream& os) const ;
-  virtual void printValue(std::ostream& os) const ;
-  void printMultiline(std::ostream& os, Int_t content, Bool_t verbose, TString indent) const ;
-
-  virtual Int_t defaultPrintContents(Option_t* opt) const ;
+  /// Print name of dataset
+  void printName(std::ostream& os) const final { os << GetName(); }
+  /// Print title of dataset
+  void printTitle(std::ostream& os) const final { os << GetTitle(); }
+  void printClassName(std::ostream& os) const final ;
+  void printArgs(std::ostream& os) const final;
+  /// Print value of the dataset, i.e. the sum of weights contained in the dataset
+  void printValue(std::ostream& os) const final { os << numEntries() << " entries" ; }
+  void printMultiline(std::ostream& os, Int_t content, Bool_t verbose, TString indent) const final;
+
+  /// Define default print options, for a given print style
+  virtual int defaultPrintContents(Option_t* /*opt*/) const final { return kName|kClassName|kArgs|kValue ; }
```

Comment:
clang-tidy will probably tell you that `virtual` and `final` are redundant here (as with override).

Reply:
I removed the `virtual` keyword, and added `const override`